src/commands/cmd_bloomfilter.cc

```diff
@@ -0,0 +1,284 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "commander.h"
+#include "error_constants.h"
+#include "server/server.h"
+#include "types/redis_bloomfilter.h"
+
+namespace redis {
+
+class CommandBFReserve : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    auto parse_error_rate = ParseFloat<double>(args[2]);
+    if (!parse_error_rate) {
+      return {Status::RedisParseErr, errValueIsNotFloat};
+    }
+    error_rate_ = *parse_error_rate;
+    if (error_rate_ >= 1 || error_rate_ <= 0) {
+      return {Status::RedisParseErr, "0 < error rate range < 1"};
+    }
+
+    auto parse_capacity = ParseInt<uint64_t>(args[3], 10);
+    if (!parse_capacity) {
+      return {Status::RedisParseErr, errValueNotInteger};
+    }
+    capacity_ = *parse_capacity;
+    if (capacity_ <= 0) {
+      return {Status::RedisParseErr, "capacity should be larger than 0"};
+    }
+
+    size_t expect_size = 4;
+    auto iter = find(args.begin(), args.end(), "NONSCALING");
```

Comment:
```
    while (parser.Good()) {
     if (parser.EatEqICaseFlag("NONSCALING", ...)) {
        ...
      } else {
        return parser.InvalidSyntax();
      }
    }
```

Using parser like this might helps